addons/spreadsheet_account/static/src/accounting_functions.js

```diff
@@ -58,11 +58,23 @@ function parseAccountingQuarter(dateRange) {
 }
 
 /**
- * @param {string} dateRange
+ * @param {object | undefined} dateRange
  * @returns {MonthDateRange | undefined}
  */
-function parseAccountingMonth(dateRange) {
-    const found = dateRange.match(MonthRegexp);
+function parseAccountingMonth(dateRange, locale) {
+    if (
+        typeof dateRange?.value === "number" &&
+        dateRange.format?.includes("m") &&
+        !dateRange.format?.includes("d")
```

Comment:
Not sure I understand what this condition is for. Why no `d` ? Are `y` fine ? I'm a bit confused

Reply:
because `parseAccountingMonth` should only handle numbers which are formatted by month. It should not handle numbers which are formatted as full dates e.g. `2024-01-01`.  `y` is fine, formats like `August 2022` is typically what we expect to match here.